### ISSUE TITLE
[Bug] Serve memory grow unbounded after hours of requests.

### ISSUE THREAD

**Issue by simon-mo** [ray-project/ray#19891]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Serve

### What happened + What you expected to happen

See https://discuss.ray.io/t/how-to-control-the-total-memory-of-ray-serve/3972

### Versions / Dependencies

OS: MacOS 10.15.7 / Linux Ubuntu 18 LTS
Ray version: 1.7.0/1.4.1
Python version: 3.7.11

### Reproduction script

https://discuss.ray.io/t/how-to-control-the-total-memory-of-ray-serve/3972

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by simon-mo** [comment]
Seems like I can repro this in ray core. I'm not 100% confident this is a real bug though: https://github.com/ray-project/ray/issues/19905

**Comment by edoakes** [comment]
Downgraded to P1 because we can't consistently repro. @simon-mo doing more investigation

**Comment by kr11** [comment]
> Pretty strange given that we ran the 2-week-long stress test w/o memory growth. Could it be an issue fixed since 1.7.0?

Hi all, thanks for your work. I re-try my script and found the mem doesn't grow in Debain. But it still increase on MacOS 10.15.7. The output is as follows:

```
(qps_prediction pid=25580) actor_pid=25580 	 2021-11-04T11:09:40Z 	 mem= 275.9 MB.
main_pid=25559 	 2021-11-04T11:09:40Z 	 mem= 264.8 MB.
main_pid=25559 	 2021-11-04T11:09:41Z 	 mem= 264.8 MB.
(qps_prediction pid=25580) actor_pid=25580 	 2021-11-04T11:09:41Z 	 mem= 285.5 MB.
main_pid=25559 	 2021-11-04T11:09:41Z 	 mem= 264.9 MB.
(qps_prediction pid=25580) actor_pid=25580 	 2021-11-04T11:09:41Z 	 mem= 293.4 MB.
main_pid=25559 	 2021-11-04T11:09:42Z 	 mem= 264.9 MB.
...
(qps_prediction pid=25580) actor_pid=25580 	 2021-11-04T11:09:47Z 	 mem= 326.1 MB.
main_pid=25559 	 2021-11-04T11:09:48Z 	 mem= 265.3 MB.
(qps_prediction pid=25580) actor_pid=25580 	 2021-11-04T11:09:48Z 	 mem= 326.2 MB.
main_pid=25559 	 2021-11-04T11:09:48Z 	 mem= 265.3 MB.
(qps_prediction pid=25580) actor_pid=25580 	 2021-11-04T11:09:48Z 	 mem= 329.2 MB.
...
```

But it doesn't matter. Our practical code is to call the PyTorch model to forecast time series.  Sadly, we found the memory still grow unbound on `Debian` + `ray-serve -1.7.0` + `pytorch_forecasting`. I will provide a public version of the code asap.

**Comment by simon-mo** [comment]
per https://discuss.ray.io/t/how-to-control-the-total-memory-of-ray-serve/3972/4 closing.
